compiler/reader.go

```diff
@@ -174,10 +175,16 @@ func ReadInfoForRef(basefile string, ref string) (interface{}, error) {
 		}
 	}
 	count = count + 1
+	basedir, _ := filepath.Split(basefile)
 	parts := strings.Split(ref, "#")
 	var filename string
 	if parts[0] != "" {
-		filename = parts[0]
+		if _, err := url.ParseRequestURI(parts[0]); err != nil {
+			// It is not an URL, so the file is local
+			filename = basedir + parts[0]
+		} else {
+			filename = parts[0]
```

Comment:
let's make this the default behavior and handle local file as a special case.

```go
filename = parts[0]
if _, err := url.ParseRequestURI(parts[0]); err != nil {
    // It is not an URL, so the file is local
    filename = basedir + parts[0]
}
```

This is code flows straight down and thus slightly more readable

Reply:
Ahhh! Done. Old habits. Thanks Noah.